app/models/rubric_criterion.rb

```diff
@@ -37,6 +38,17 @@ def update_assigned_groups_count
     self.assigned_groups_count = result.uniq.length
   end
 
+  def scale_marks_if_max_mark_changed
+    max_level_mark = self.levels.max_by(&:mark).mark
+    new_max = self.max_mark
+    return if max_level_mark == new_max
+
+    scale = new_max / max_level_mark
+    self.levels.each do |level|
+      level.mark = (level.mark * scale).round(1)
```

Comment:
You'll want to do an `level.update` here to make sure the changes are saved. Also round to 2 decimal places.